kernel/cpu/src/aarch64.rs

```diff
@@ -1,27 +1,10 @@
+//! Implements functions for accessing CPU-specific information on aarch64.
+
 use cortex_a::registers::MPIDR_EL1;
 use tock_registers::interfaces::Readable;
 
 use core::fmt;
 
-/// A unique identifier for a CPU core.
-#[derive(Copy, Clone, Debug, PartialEq, Eq)]
-#[repr(transparent)]
-pub struct CpuId(u32);
-
-/// An equivalent to Option<CpuId>, which internally encodes None as
-/// `u32::MAX`, which is an invalid CpuId (bits [4:7] of affinity level
-/// 0 must always be cleared). This guarantees that it compiles down to
-/// lock-free native atomic instructions when using it inside of an atomic
-/// type like [`AtomicCell`], as u32 is atomic when running on ARMv8.
-#[derive(Clone, Copy, PartialEq, Eq)]
-#[repr(transparent)]
-pub struct OptionalCpuId(u32);
-
-/// A unique identifier for a CPU core.
-#[derive(Copy, Clone, Debug)]
-#[repr(transparent)]
-pub struct MpidrValue(u64);
-
```

Comment:
I personally think people should first learn about the types when they open a file, which is why I was placing these definitions here. Was there a motivation for moving them lower?

Also, CpuId is imported from `super` so using it shouldn't compile?

Reply:
it's more canonical to group a type's definition with its implementations, otherwise they can be hard to find. The only place you can't do that is when the impls must be in different modules, e.g., due to different target archs.

> Also, CpuId isn't imported from `super` so using it shouldn't compile?
Huh, that is indeed weird.

Reply:
The `cpu` crate is not included in an aarch64 build, that's why. I'll let you deal with that in a future PR.